src/commands/local.ts

```diff
@@ -758,6 +760,31 @@ function reportSuccess(
     functionUrl: `http://127.0.0.1:${port}/${c.name}`,
   }));
 
+  // get installed version of node
+  const nodeVersion = process.version;
+
+  // get only the major version
+  const nodeMajorVersion = nodeVersion.split(".")[0].slice(1);
+
+  // get server used version
+  let serverRuntime: string = DEFAULT_NODE_RUNTIME as string;
+  if (projectConfiguration.options?.nodeRuntime) {
+    serverRuntime = projectConfiguration.options.nodeRuntime;
+  }
+
+  const serverVersion = serverRuntime.split(".")[0].split("nodejs")[1];
+
+  log.debug(`Node version: ${nodeVersion}`);
+  log.debug(`Server version: ${serverRuntime}`);
+
+  // check if server version is different from installed version
+  if (nodeMajorVersion !== serverVersion) {
+    log.warn(`${colors.yellow(`Warning: You are using node version ${nodeVersion} but your server is using ${serverRuntime}. This might cause unexpected behavior.
```

Comment:
This warning appears even if I don't have a project deployed. This might be confusing to the user because "what server when I deployed nothing".

Maybe we can tweak the message a bit to `but your server is configured to use ${serverRuntime}`